app/cdap/components/RecipeManagement/CreateRecipe/index.tsx

```diff
@@ -0,0 +1,221 @@
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { FormEvent, useEffect, useRef, useState } from 'react';
+import { debounce } from 'lodash';
+import T from 'i18n-react';
+import { ActionType } from 'components/RecipeList/types';
+import {
+  ICreateRecipeProps,
+  IRecipeFormData,
+  IRecipeNameErrorData,
+} from 'components/RecipeManagement/types';
+import RecipeForm from 'components/RecipeManagement/RecipeForm';
+import MyDataPrepApi from 'api/dataprep';
+import { getCurrentNamespace } from 'services/NamespaceStore';
+import useFetch from 'services/react/customHooks/useFetch';
+
+const PREFIX = 'features.WranglerNewUI.RecipeForm.labels';
+const recipeNameRegEx = /^[a-z\d\s]+$/i;
+export const noErrorState: IRecipeNameErrorData = {
+  isRecipeNameError: false,
+  recipeNameErrorMessage: '',
+};
+
+export default function CreateRecipe({ setShowRecipeForm, setSnackbar }: ICreateRecipeProps) {
+  const [recipeFormData, setRecipeFormData] = useState({
+    recipeName: '',
+    description: '',
+  });
+  const [recipeNameErrorData, setRecipeNameErrorDataState] = useState(noErrorState);
+
+  const [isSaveDisabled, setIsSaveDisabled] = useState(true);
+  const [apiParams, setApiParams] = useState({
+    getRecipeByNameParams: {
+      context: '',
+      recipeName: '',
+    },
+    createRecipeParams: {
+      context: '',
+    },
+  });
+
+  /*
+   * TODO: This static data has to be removed when we have actual API data,
+   * then directly we will get that data from store as directives
+   */
+
+  const recipeSteps = ['uppercase: body1', 'titlecase: body2'];
+
+  const { response: recipeByNameResponse, error: recipeByNameError } = useFetch(
+    MyDataPrepApi.getRecipeByName,
+    apiParams.getRecipeByNameParams,
+    'getRecipeByName'
+  );
+  const { response: createRecipeResponse, error: createRecipeError } = useFetch(
+    MyDataPrepApi.createRecipe,
+    apiParams.createRecipeParams,
+    'createRecipe',
+    {
+      recipeName: recipeFormData.recipeName,
+      description: recipeFormData.description,
+      directives: recipeSteps,
+    }
+  );
+
+  useEffect(() => {
+    if (recipeByNameResponse) {
+      setRecipeNameErrorData(
+        {
+          isRecipeNameError: true,
+          recipeNameErrorMessage: T.translate(`${PREFIX}.sameNameErrorMessage`).toString(),
+        },
+        recipeFormData
+      );
+    }
+    if (recipeByNameError) {
+      if (recipeByNameError.statusCode === 404) {
+        setRecipeNameErrorData(noErrorState, recipeFormData);
+      } else {
+        setSnackbar({
+          open: true,
+          isSuccess: false,
+          message: (recipeByNameError.response as Record<string, string>).message,
+        });
+      }
+    }
+  }, [recipeByNameResponse, recipeByNameError]);
+
+  useEffect(() => {
+    if (createRecipeResponse) {
+      setShowRecipeForm(false);
```

Comment:
`setShowRecipeForm(false);`

this seems to be the default behavior when both dependencies are updating
plz move it out of conditional blocks and also mention the use case here

Reply:
here after code updates we've state update only once without any conditional check